public/src/components/channelManagement/campaigns/TestDataDialog.tsx

```diff
@@ -0,0 +1,537 @@
+import React from 'react';
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  IconButton,
+  makeStyles,
+} from '@material-ui/core';
+import CloseIcon from '@material-ui/icons/Close';
+import { Test } from '../helpers/shared';
+
+const useStyles = makeStyles(() => ({
+  dialogHeader: {
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'space-between',
+    paddingRight: '8px',
+  },
+  input: {
+    '& input': {
+      textTransform: 'uppercase !important',
+    },
+  },
+  buttonContainer: {
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'flex-end',
+  },
+  testDataContainer: {
+    fontFamily: 'monospace',
+    fontSize: '12px',
+    whiteSpace: 'pre-wrap',
+  },
+}));
+
+const testFields = {
+  core: {
+    name: {
+      label: 'Tracking name',
+      type: 'string',
+      exclude: [],
+      optional: false,
+    },
+    nickname: {
+      label: 'Nickname (RRCP)',
+      type: 'string',
+      exclude: [],
+      optional: true,
+    },
+    channel: {
+      label: 'Marketing channel',
+      type: 'string',
+      exclude: [],
+      optional: true,
+    },
+    status: {
+      label: 'Test status',
+      type: 'string',
+      exclude: [],
+      optional: false,
+    },
+    priority: {
+      label: 'Channel priority',
+      type: 'number',
+      exclude: [],
+      optional: true,
+    },
+    campaignName: {
+      label: 'Campaign',
+      type: 'string',
+      exclude: [],
+      optional: true,
+    },
+    highPriority: {
+      label: 'Is high priority (deprecated)',
+      type: 'boolean',
+      exclude: ['Header', 'Banner1', 'Banner2'],
+      optional: true,
+    },
+  },
+
+  targeting: {
+    userCohort: {
+      label: 'User cohort',
+      type: 'string',
+      exclude: [],
+      optional: true,
+    },
+    locations: {
+      label: 'Locations',
+      type: 'string-array',
+      exclude: [],
+      optional: false,
+    },
+    deviceType: {
+      label: 'Device type',
+      type: 'string',
+      exclude: ['Banner1', 'Banner2'],
```

Comment:
There is a device type on banners, is it not actually used?

Reply:
You're right. Good spot!